pilot/cmd/pilot-agent/main.go

```diff
@@ -586,7 +588,16 @@ func getMeshConfig() (meshconfig.MeshConfig, error) {
 		}
 		return *mc, nil
 	}
-	return defaultConfig, nil
+	b, err := ioutil.ReadFile(meshConfigFile)
+	if err != nil {
+		log.Warnf("Failed to read mesh config file from %v or MESH_CONFIG. Falling back to defaults: %v", meshConfigFile, err)
+		return defaultConfig, nil
+	}
+	mc, err := mesh.ApplyMeshConfig(string(b), defaultConfig)
+	if err != nil || mc == nil {
+		return meshconfig.MeshConfig{}, fmt.Errorf("failed to unmarshal mesh config config: %v", err)
```

Comment:
Can you print the yaml here along with the error? Same in meshconfig unmarshall log line

Reply:
It may be more idiomatic that this function returns a pointer and we return nil,err here and other place instead of returning empty MeshConfig

Reply:
I added this log